doc/style-guide.md

```diff
@@ -91,6 +92,103 @@ quaility.
 
 **[top](#table-of-contents)**
 
+## Object and Array Literals
+
+  - Object literals
+    - Use curly braces `{}` instead of `new Object()`.
+    - The opening curly brace (i.e., `{`) must be on the same line as the declaration, expression,
+      or statement.
+    - The closing curly brace (i.e., `}`) must be on its own line aligned at the same indentation
+      level as line with the opening curly brace.
+    - Each `property: value` pair must be on its own line followed by a comma `,` indented by the
+      [indentation](#indentation) style specified within this document.
+    - The final property/value pair *should not* be followed by a comma `,`.
+    - If a object contains at most one property/value pair, it acceptable to have the
+      property/value pair, if it exists, on one line with a space after the opening `{` and before
+      the closing `}`.
+    ```typescript
+    // bad
+    var obj = {foo: 1};
+
+    // bad
+    var obj = { foo: 1, bar: 2 };
+
+    // bad
+    var obj = { foo: 1,
+                bar: 2};
+
+    // bad
+    var obj = {
+        foo: 1,
+        bar: 2,
+    };
+
+    // OK
+    var obj = { foo: 1 };
+
+    // good
+    var obj = {};
+
+    // good
+    var obj = {
+        foo: 1
+    };
+
+    // good
+    var obj = {
+        foo: 1,
+        bar: 2
+    };
+    ```
+
+  - Array literals
+    - Use brackets `[]` instead of `new Array()`.
+    - The opening bracket (i.e., `[`) must be on the same line as the declaration, expression, or
+      statement.
+    - The closing bracket (i.e., `]`) must be on its own line aligned at the same indentation level
+      as line with the opening bracket.
+    - Each array element must be on its own line followed by a comma `,` indented by the
+      [indentation](#indentation) style specified within this document.
+    - The final array element *should not* be followed by a comma `,`.
+    - It is acceptable to have an array literal on one line if and only if it fits within
```

Comment:
I will vote for this as the default style for array literal(if not the only option) since it is the most common use case. One line per element seems a bit weird to me. @yupinghu, thoughts?